transformer_engine/jax/transformer.py

```diff
@@ -0,0 +1,917 @@
+    dtype :jax.numpy.dtype, default  = jax.numpy.float32
+        the data type used to allocate the initial parameters.
+    fuse_qkv: bool, default = True
+        if set to True, this module exposes a single fused
+        parameter for query-key-value for self-attention and key-value for
+        cross-attention.
+    transpose_batch_sequence : bool, default = True
+        indicate whether the input tensors were switched axis of batch
+        and sequence length dimension. if set to True, the input tensors
+        should be in (seqlen, batch, hidden), otherwise (batch, seqlen, hidden).
+    scale_attn_logits: bool, default = False
+        indicate whether to scale attention logits.
+        if set to True, :math:`\frac{Q}{\sqrt{head_dim}*K}`,
+        else :math:`Q*K`
+    scaled_query_init: bool, default = `True`
+        whether to scale WQ on initilization by :math:`\sqrt{head_dim}`
+    float32_logits : bool, default = False
+        whether to compute attention logits in float32.
+    """
+
+    head_dim: int
+    num_heads: int
+    dropout_rate: float = 0.
+    dropout_rng_name: str = 'dropout'
+    layernorm_type: str = "layernorm"
+    layernorm_epsilon: float = 1e-6
+    kernel_init: Initializer = nn.initializers.variance_scaling(1.0, 'fan_in', 'normal')
+    use_bias: bool = False
+    bias_init: Initializer = nn.initializers.zeros
+    apply_residual_connection_post_layernorm: bool = False
+    output_layernorm: bool = False
+    attn_type: AttentionType = AttentionType.PADDING
+    dtype: DType = jnp.float32
+    fuse_qkv: bool = True
+    transpose_batch_sequence: bool = True
+    scale_attn_logits: bool = False
+    scaled_query_init: bool = True
+    float32_logits: bool = False    # computes logits in float32 for stability.
+
+    @nn.compact
+    def __call__(self,
+                 inputs_q: Array,
+                 inputs_kv: Array,
+                 mask: Optional[Array] = None,
+                 bias: Optional[Array] = None,
+                 *,
+                 decode: bool = False,
+                 deterministic: bool = False) -> Array:
+        """Applies multi-head dot product attention on the input data."""
+
+        depth_scaling = jnp.sqrt(self.head_dim).astype(self.dtype)
+
+        def query_init(*args):
+            return self.kernel_init(*args) / (depth_scaling if self.scaled_query_init else 1.0)
+
+        def qkv_init(key, shape, dtype):
+            assert len(shape) == 3
+            assert shape[-2] == 3
+
+            q_key, k_key, v_key = jax_random.split(key, num=3)
+
+            q_shape = (shape[0], shape[-1])
+            k_shape = (shape[0], shape[-1])
+            v_shape = (shape[0], shape[-1])
+
+            q_kernel = query_init(q_key, q_shape, dtype)
+            k_kernel = self.kernel_init(k_key, k_shape, dtype)    # pylint: disable=too-many-function-args
+            v_kernel = self.kernel_init(v_key, v_shape, dtype)    # pylint: disable=too-many-function-args
```

Comment:
```suggestion
            k_kernel = self.kernel_init(k_key, k_shape, dtype)
            v_kernel = self.kernel_init(v_key, v_shape, dtype)
```

Reply:
@jeng1220 Please review, I'm unable to repro this lint error locally and it passes without these disables

Reply:
ok, I removed the `disable=too-many-function-args` in afa3dc2